src/main/java/ca/attractors/chess/MoveDirection.java

```diff
@@ -58,17 +58,20 @@ public static MoveDirection getMoveDirection(Position start, Position end) {
     }
 
     private static Stream<MoveDirection> getMoveDirections(int horizontalShift, int verticalShift) {
-        MoveDirection[] moveDirections = null;
-        if (horizontalShift == 0) {
-            moveDirections = VERTICAL_ONLY;
-        } else if (verticalShift == 0) {
-            moveDirections = HORIZONTAL_ONLY;
-        } else if (Math.abs(horizontalShift) == Math.abs(verticalShift)) {
-            moveDirections = DIAGONAL_ONLY;
-        }
+        MoveDirection[] moveDirections = getMoveDirectionsArray(horizontalShift, verticalShift);
         return (moveDirections != null) ? Arrays.stream(moveDirections) : Stream.empty();
     }
 
+    private static MoveDirection[] getMoveDirectionsArray(int horizontalShift, int verticalShift) {
```

Comment:
Yeah, this totally makes sense. I missed that rule you presented during the training 😮